From 6608a53ebcd3ca592726915c538cfe4e342d3bc1 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:20 -0400
Subject: [block] blk-mq: add file comments and update copyright notices

Message-id: <1402670238-13916-93-git-send-email-snitzer@redhat.com>
Patchwork-id: 83819
O-Subject: [RHEL7.1 PATCH 092/150] blk-mq: add file comments and update copyright notices
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 75bb4625bb78d6a2d879dcb6a7d482861295765b
Author: Jens Axboe <axboe@fb.com>
Date:   Wed May 28 10:15:41 2014 -0600

    blk-mq: add file comments and update copyright notices

    None of the blk-mq files have an explanatory comment at the top
    for what that particular file does. Add that and add appropriate
    copyright notices as well.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq-cpu.c b/block/blk-mq-cpu.c
index d2c253f..bb3ed48 100644
--- a/block/blk-mq-cpu.c
+++ b/block/blk-mq-cpu.c
@@ -1,3 +1,8 @@
+/*
+ * CPU notifier helper code for blk-mq
+ *
+ * Copyright (C) 2013-2014 Jens Axboe
+ */
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/init.h>
diff --git a/block/blk-mq-cpumap.c b/block/blk-mq-cpumap.c
index 0daacb9..1065d7c 100644
--- a/block/blk-mq-cpumap.c
+++ b/block/blk-mq-cpumap.c
@@ -1,3 +1,8 @@
+/*
+ * CPU <-> hardware queue mapping helpers
+ *
+ * Copyright (C) 2013-2014 Jens Axboe
+ */
 #include <linux/kernel.h>
 #include <linux/threads.h>
 #include <linux/module.h>
diff --git a/block/blk-mq-tag.c b/block/blk-mq-tag.c
index 0d0640d..d90c4ae 100644
--- a/block/blk-mq-tag.c
+++ b/block/blk-mq-tag.c
@@ -1,3 +1,15 @@
+/*
+ * Fast and scalable bitmap tagging variant. Uses sparser bitmaps spread
+ * over multiple cachelines to avoid ping-pong between multiple submitters
+ * or submitter and completer. Uses rolling wakeups to avoid falling of
+ * the scaling cliff when we run out of tags and have to start putting
+ * submitters to sleep.
+ *
+ * Uses active queue tracking to support fairer distribution of tags
+ * between multiple submitters when a shared tag map is used.
+ *
+ * Copyright (C) 2013-2014 Jens Axboe
+ */
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/random.h>
diff --git a/block/blk-mq.c b/block/blk-mq.c
index ae14749..5cc4b87 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1,3 +1,9 @@
+/*
+ * Block multiqueue core code
+ *
+ * Copyright (C) 2013-2014 Jens Axboe
+ * Copyright (C) 2013-2014 Christoph Hellwig
+ */
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/backing-dev.h>
-- 
1.7.1